src/authority.rs

```diff
@@ -46,3 +47,32 @@ impl DocumentAuthority for FileSystemDocumentAuthority {
         Ok(schema_content)
     }
 }
+
+/// An [Authority] implementation that attempts to resolve schema ids to ion elements using the map.
+#[derive(Debug, Clone)]
+pub struct MapDocumentAuthority {
+    ion_content_by_id: HashMap<String, String>, // This map represents (id, ion content) which can used to resolve schema ids to Vec<OwnedElement>
+}
```

Comment:
Nice! I like that storing this as a `HashMap<String, String>` allows us to write tests that fail because the Ion/ISL that was stored in the map was invalid. Good thinking!